docs(splitter): Revamp Panes article (#2241)
telerik/blazor-docs: components/splitter/panes.md

| `Size` | `string` | The pane `width` CSS style in [horizontal Splitters]({%slug splitter-orientation%}), or the pane `height` in [vertical Splitters]({%slug splitter-orientation%}). Supports two-way binding. The `Size` must be between the `Min` and `Max` values. See [Pane Size](#pane-size) below for more details on pane dimensions and behavior. |
| `Visible` | `bool` | Defines if the pane element and splitbar render or not. When toggled at runtime, the pane's index remains unchanged, unlike when adding a pane with a conditional statement, which appends it at the end. Compare with the `Collapsed` parameter. |

>caption Configure Splitter Panes

````CSHTML
<TelerikSplitter Width="600px" Height="300px">
    <SplitterPanes>

        <SplitterPane Collapsible="true" @bind-Size="@PaneSize1" Min="100px" Max="300px">
            <h4>Left Pane</h4>
            <div>Collapsible and scrollable pane with initial size in px. It can be resized between 100px and 300px.</div>
        </SplitterPane>

        <SplitterPane Collapsible="true" @bind-Size="@PaneSize2">
            <h4>Middle Pane</h4>

>             <div>Collapsible pane with initial size in per cent.</div>

Reviewer: ```suggestion
            <div>Collapsible pane with initial size in percentage.</div>
```

        <SplitterPane Visible="false">
            <h4>Invisible Pane</h4>
            <span>This pane will not render.</span>
        </SplitterPane>

    </SplitterPanes>
</TelerikSplitter>

@code {
    private string PaneSize1 { get; set; } = "200px";

    private string PaneSize2 { get; set; } = "20%";
}
````

## Pane Scrolling

To make a Splitter pane scrollable without using additional HTML markup or custom CSS styles, set the pane `Class` parameter to `k-scrollable`. See the example above.

## Pane Dimensions

The dimensions of a Splitter pane depend on:

* The [pane `Size`](#pane-size) parameter
* The [pane `Collapsible` and `Resizable`](#pane-collapsibility-and-resizability) parameters
* The [Splitter `Width`, `Height`, and `Orientation`](#splitter-width-and-height) parameters

The sections below provide more details and a [hands-on example](#example).

### Pane Size

The Splitter pane `Size` controls the pane width or height, depending on the [Splitter `Orientation`]({%slug splitter-orientation%}).

There must be at least one `SplitterPane` without a `Size`. This pane will adjust automatically to occupy the remaining space, based on the other pane sizes.

> If the pane `Size` is greater than `Max`, the pane cannot be resized even if its `Resizable` parameter is set to `true`.

Reviewer: That's weird. I wouldn't expect it to be expandable, but why can't you make it smaller?

Author: The `Size` can be larger than `Max` only if the developer sets it like this. The user can never achieve such a state on their own. We consider it an invalid configuration, so we don't invest dev effort to fix it.

### Pane Collapsibility and Resizability

Collapsibility and resizability have the following impact on the Splitter pane dimensions:

* Panes that are collapsible or resizable are called *flex panes*. When a flex pane has no `Size`, it expands to fill the available space. If multiple flex panes have no `Size`, they take up equal parts of the available space.

> * Panes that are not collapsible and not resizable are called *static panes*. When a static pane has no `Size`, it expand and shrinks based on its content.

Reviewer: ```suggestion
* Panes that are not collapsible and not resizable are called *static panes*. When a static pane has no `Size`, it expands and shrinks based on its content.
```

### Splitter Width and Height

In a [vertical Splitter]({%slug splitter-orientation%}), the pane widths match the Splitter `Width`.

Here is how the Splitter `Height` affects the pane heights:

> * If a [horizontal Splitter]({%slug splitter-orientation%}) has no `Height`, then its panes do not expand vertically to fill up the Splitter element. The [example](#example) below shows how to workaround this with a `height:auto` style on the `.k-pane` class.

Reviewer: Usually, `workaround` is the noun or adjective form, while `work around` would be the verb.

        FirstPaneSizeNumber = newValue;

        if (FirstPaneSizeNumber.HasValue)
        {
            FirstPaneSize = $"{FirstPaneSizeNumber}%";
        }
        else
        {
            FirstPaneSize = string.Empty;
        }
    }

    private void LastPaneSizeNumberChanged(int? newValue)
    {
        LastPaneSizeNumber = newValue;

        if (LastPaneSizeNumber.HasValue)
        {
            LastPaneSize = $"{LastPaneSizeNumber}%";
        }
        else
        {
            LastPaneSize = string.Empty;
        }
    }

    private void FirstPaneSizeChanged(string newValue)
    {
        newValue = newValue.Replace("%", "").Replace("px", "");
        FirstPaneSizeNumberChanged(Convert.ToInt32(Math.Round(Convert.ToDecimal(newValue))));
    }

    private void LastPaneSizeChanged(string newValue)
    {
        newValue = newValue.Replace("%", "").Replace("px", "");
        LastPaneSizeNumberChanged(Convert.ToInt32(Math.Round(Convert.ToDecimal(newValue))));
    }

    protected override void OnInitialized()
    {
        FirstPaneSizeNumberChanged(FirstPaneSizeNumber);
        LastPaneSizeNumberChanged(LastPaneSizeNumber);

        base.OnInitialized();
    }
}
````

> ## Nest Steps

Reviewer: ```suggestion
## Next Steps
```
